File: frontend/src/types/analysis.ts
export interface AnalysisCapabilities {
  configured_provider: string;
  provider_available: boolean;
  remote_provider: boolean;
  remote_consent_required: boolean;
  extraction_version: string;
  confidence_version: string;
  max_conversations_per_request: number;
}

export interface AnalysisErrorRecord {
  error_code: string;
  message: string;
  recoverable: boolean;
  insight_id?: string;
  conversation_id?: string;
  window_id?: string;
}

export interface AnalysisResponse {
  request_id: string;
  provider_name: string;
  conversation_count: number;
  selected_message_count: number;
  window_count: number;
  successful_window_count: number;
  failed_window_count: number;
  candidates_received: number;
  candidates_accepted: number;
  insights_created: number;
  insights_reused: number;
  insight_ids: string[];
  confidence_scored_count: number;
  confidence_failed_count: number;
  stopped_early: boolean;
  errors: AnalysisErrorRecord[];
  insights_link: string;
}

export interface StartAnalysisInput {
  conversationIds: string[];
  remoteConsent: boolean;
}

File: backend/src/echomind/schemas/analysis.py
"""Bounded public contracts for synchronous user-triggered analysis."""

from typing import Annotated
from uuid import UUID

from pydantic import AwareDatetime, Field, field_validator, model_validator

from echomind.schemas.common import StrictSchema


class AnalysisRequest(StrictSchema):
    conversation_ids: Annotated[list[UUID], Field(min_length=1, max_length=100)]
    remote_consent: bool = False
    start_at: AwareDatetime | None = None
    end_at: AwareDatetime | None = None
    stop_on_window_error: bool = True

    @field_validator("conversation_ids")
    @classmethod
    def stable_deduplicate_ids(cls, values: list[UUID]) -> list[UUID]:
        return list(dict.fromkeys(values))

    @model_validator(mode="after")
    def time_range_is_valid(self) -> "AnalysisRequest":
        if self.start_at is not None and self.end_at is not None and self.end_at < self.start_at:
            raise ValueError("end_at must not be earlier than start_at")
        return self


class AnalysisCapabilities(StrictSchema):
    configured_provider: str
    provider_available: bool
    remote_provider: bool
    remote_consent_required: bool
    extraction_version: str
    confidence_version: str
    max_conversations_per_request: int


class AnalysisErrorRecord(StrictSchema):
    error_code: str
    message: str
    recoverable: bool
    insight_id: str | None = None
    conversation_id: str | None = None
    window_id: str | None = None


class AnalysisResponse(StrictSchema):
    request_id: UUID
    provider_name: str
    conversation_count: int = Field(ge=0)
    selected_message_count: int = Field(ge=0)
    window_count: int = Field(ge=0)
    successful_window_count: int = Field(ge=0)
    failed_window_count: int = Field(ge=0)
    candidates_received: int = Field(ge=0)
    candidates_accepted: int = Field(ge=0)
    insights_created: int = Field(ge=0)
    insights_reused: int = Field(ge=0)
    insight_ids: list[str]
    confidence_scored_count: int = Field(ge=0)
    confidence_failed_count: int = Field(ge=0)
    stopped_early: bool
    errors: list[AnalysisErrorRecord]
    insights_link: str
